entity-framework/core/managing-schemas/migrations/index.md

```diff
@@ -0,0 +1,198 @@
+    table: "Customer");
+```
+
+> [!TIP]
+> Adding a new migration warns when an operation is scaffolded that may result in data loss (like dropping a column). Be
+> sure to especially review these migrations for accuracy.
+
+Apply the migration to the database using the appropriate command.
+
+``` powershell
+Update-Database
+```
+``` Console
+dotnet ef database update
+```
+
+Removing a migration
+--------------------
+Sometimes you add a migration and realize you need to make additional changes to your EF Core model before applying it.
+To remove the last migration, use this command.
+
+``` powershell
+Remove-Migration
+```
+``` Console
+dotnet ef migrations remove
+```
+
+After removing it, you can make the additional model changes and add it again.
+
+Reverting a migration
+---------------------
+If you already applied a migration (or several migrations) to the database but need to revert it, you can use the same
+command to apply migrations, but specify the name of the migration you want to roll back to.
+
+``` powershell
+Update-Database LastGoodMigration
+```
+``` Console
+dotnet ef database update LastGoodMigration
+```
+
+Empty migrations
+----------------
+Sometimes it's useful to add a migration without making any model changes. In this case, adding a new migration creates
+an empty one. You can customize this migration to perform operations that don't directly relate to the EF Core model.
+Some things you might want to manage this way are:
+
+* Full-Text Search
+* Functions
+* Stored procedures
+* Triggers
+* Views
+* etc.
+
+Generating a SQL script
+-----------------------
+When debugging your migrations or deploying them to a production database, it's useful to generate a SQL script. The
+script can then be further reviewed for accuracy and tuned to fit the needs of a production database. The script can
+also be used in conjunction with a deployment technology. The basic command is as follows.
+
+``` powershell
+Script-Migration
+```
+``` Console
+dotnet ef migrations script
+```
+
+There are several options to this command.
+
+The **from** migration should be the last migration applied to the database before running the script. If no migrations
+have been applied, specify `0` (this is the default).
+
+The **to** migration is the last migration that will be applied to the database after running the script. This defaults
+to the last migration in your project.
+
+An **idempotent** script can optionally be generated. This script only applies migrations if they haven't already been
+applied to the database. This is useful if you don't exactly know what the last migration applied to the database was or
+if you are deploying to multiple databases that may each be at a different migration.
+
+Applying migrations at runtime
+------------------------------
+Some apps may want to apply migrations at runtime during startup or first run. Do this using the `Migrate()` method.
+
+``` csharp
+myDbContext.Database.Migrate();
+```
+
+> [!WARNING]
```

Comment:
I think we should be more explicit about not running migrations at application startup except in very specific cases, such as testing or client apps with a local database. The advice we have generally been giving is that even if you want to use the update-database command on your production database it should only be done in a controlled environment as part of a controlled application deployment.

Reply:
Updated.